Lecture6/Task1/HttpClientsImplementation/ApacheHttpClient.java

```diff
@@ -0,0 +1,62 @@
+package HttpClientsImplementation;
+
+import java.io.IOException;
+
+import HttpClientsImplementation.Utils.RequestDataPreparatory;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthenticationException;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.impl.auth.BasicScheme;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+public class ApacheHttpClient {
+
+    private static final String ERROR_MESSAGE_IO_EXCEPTION = "Something went wrong (ApacheHttpClient => IOException)";
+    private static final String ERROR_MESSAGE_AUTH_EXCEPTION = "Something went wrong (ApacheHttpClient => AuthenticationException)";
+
+    private ApacheHttpClient() {
```

Comment:
is it vital to restrict creating instances of this client?

Reply:
fixed